shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java

```diff
@@ -277,7 +289,11 @@ private void setSystemChromeSystemUIOverlayStyle(
   }
 
   private void popSystemNavigator() {
-    activity.finish();
+    if (fragmentActivity != null) {
```

Comment:
Rather than creating a new constructor and field, this could just be `activity instanceof FragmentActivity`